File: ruby/engine/spec/tic_tac_toe/human_spec.rb
require 'spec_helper'

describe TakaTicTacToe::Human do
  let(:human) {human = described_class.new('X')}
  describe "#new" do
    it "should create an object player with a mark" do
      human.mark.should eq('X')
    end
  end

  it "should make a move on the board" do
    board = game_board.new
    human.make_move(1, board)
  end
end

File: objective-c/console/ttt_engine/Specs/HumanSpec.m
#import <OCDSpec2/OCDSpec2.h>
#import "HumanPlayer.h"
#import "MockUi.h"
#import "board.h"

OCDSpec2Context(HumanSpec) {

  Describe(@"Human Player", ^{
      __block HumanPlayer *humanPlayer;

      BeforeEach(^{
          humanPlayer = [[HumanPlayer alloc] init:@"X"];
      });

    It(@"has a mark for Human Player", ^{
        [ExpectStr(humanPlayer.mark) toBe:@"X"];
    });

      It(@"makes a move", ^{
          Board* board = [[Board alloc] init:3];
          MockUi* mockUi = [[MockUi alloc] init];

          [humanPlayer makeMoveWith:board andUi:mockUi];

          [ExpectBool(mockUi.promptedMove) toBeTrue];
          [ExpectStr([board.slots objectAtIndex:4]) toBe:@"X"];
      });
  });
}
